### ISSUE TITLE
null-ls won't start (eslint, prettierd)

### ISSUE THREAD

**Issue by imjamesdotme** [LunarVim/LunarVim#3737]
### Problem description

I have just updated LunarVim for the first time since around late November last year (love a lot of the new changes) but as part of the updates null-ls is no longer starting when I open Typescript & Javascript files. Additionally, I'm not able to format manually using `lf`.

I have the following linter & formatter setup in my LunarVim config file.
```
-- Linting
local linters = require "lvim.lsp.null-ls.linters"
linters.setup {
  {
    command = "eslint",
    fileTypes = "typescript, typescriptreact, javascript"
  }
}

local formatters = require "lvim.lsp.null-ls.formatters"
formatters.setup {
  {
    command = 'prettierd',
    fileTypes = { "typescript", "typescriptreact", "javascript", "css", "html" }
  }
}
```

Output from `LvimInfo`
```

    __                          _    ___         
   / /   __  ______  ____ _____| |  / (_)___ ___ 
  / /   / / / / __ \/ __ `/ ___/ | / / / __ `__ \
 / /___/ /_/ / / / / /_/ / /   | |/ / / / / / / /
/_____/\__,_/_/ /_/\__,_/_/    |___/_/_/ /_/ /_/ 


Buffer info
* filetype:                typescriptreact
* bufnr:                   4
* treesitter status:       active

Active client(s)
* name:                      tsserver
* id:                        1
* supported filetype(s):     [javascript, javascriptreact, javascript.jsx, typescript, typescriptreact, typescript.tsx]
* attached buffers:          4
* root_dir pattern:          4
* capabilities:              documentSymbolProvider          | hoverProvider             | workspaceSymbolProvider
                             typeDefinitionProvider          | foldingRangeProvider      | documentFormattingProvider
                             implementationProvider          | documentHighlightProvider | inlayHintProvider
                             documentRangeFormattingProvider | definitionProvider
                             selectionRangeProvider          | referencesProvider

Automatic LSP info
* Skipped servers: [angularls, ansiblels, csharp_ls, cssmodules_ls, denols, ember, emmet_ls, eslint, glint, golangci_lint_ls, gradle_ls, graphql, jedi_language_server, ltex, neocmake, phpactor, psalm, pylsp, quick_lint_js, reason_ls, rnix, rome, ruby_ls, ruff_lsp, solang, solc, sorbet, sourcery, spectral, sqlls, sqls, stylelint_lsp, svlangserver, tflint, verible, vuels]

Formatters info
* Active: prettierd  
* Supported: [deno_fmt, dprint, eslint, eslint_d, prettier, prettier_d_slim, prettier_eslint, prettierd, rome, rustywind, standardts]

Linters info
* Active: eslint  
* Supported: [eslint, eslint_d, semgrep, tsc, xo]

Code actions info
* Active: 
```

Output from `NullLsInfo`:
```
	null-ls
	https://github.com/jose-elias-alvarez/null-ls.nvim
	
	Logging
	* current level: warn
	* path: /<REMOVED>/<REMOVED>/.cache/lvim/null-ls.log
	
	Active source(s)
	* name: eslint
	* filetypes: vue | javascriptreact | javascript | typescript | typescriptreact
	* methods: diagnostics
	
	* name: prettierd
	* filetypes: yaml | typescriptreact | handlebars | jsonc | typescript | markdown.mdx | html | markdown | vue | javascriptreact | less | graphql | scss | json | css | javascript
	* methods: formatting
	
	
	Supported source(s)
	* code_actions: eslint | eslint_d | xo
	* formatting: deno_fmt | dprint | eslint | eslint_d | prettier | prettier_d_slim | prettier_eslint | prettierd | rome | rustywind | standardts
	* diagnostics: eslint | eslint_d | semgrep | tsc | xo
```

I'm able to replicate this issue on two different Mac's using the same versions of Neovim & LunarVim. I'm unable to diagnose any further, any help would be much appreciated.

### LunarVim version

master-4b77c9c

### Neovim version (>= 0.8.0)

0.8.2

### Operating system/version

macOS 12.2.1

### Steps to reproduce

1. Open Javascript or Typescript file.
2. Press `l i` for LSP info.
3. Observe null-ls is set to `autostart: false`.

### Screenshots

![Screenshot 2023-01-16 at 09 32 30](https://user-images.githubusercontent.com/9949268/212645358-2be8a760-1b43-4a88-99d7-5ef3282dfce9.png)

**Comment by opalmay** [comment]
I just noticed today that I have the exact same thing with python, black formatter and flake8 linter. 
I'm sure it's worked a few days ago (I'm keeping up to date with master), but even reverting to older commits and resyncing plugins didn't work, and I don't see any recent null-ls change.
btw, `autostart: false` is normal for null-ls, and all other indicators are normal as you stated. Still, lint is not showing and the code is not formatting.
@LostNeophyte  Any idea?

**Comment by opalmay** [comment]
Eerily I just noticed today that I have the exact same thing with python, black formatter and flake8 linter. 
However, it doesn't seem to be related to any recent change, as there weren't any null-ls related changes in the last few days, and going back to older commits and even to stable with a minimal config of just setting up black formatter it still wouldn't work.
I'm also running 0.8.2

btw, `autostart: false` is normal for null-ls, and all other indicators are normal as you stated. Still, lint is not showing and the code is not formatting.

@LostNeophyte  Any idea?

**Comment by opalmay** [comment]
Can you try to see if you can replicate this on stable too now? If not, we are probably not having the same root cause at all and I won't steal your issue lol
You can do it without reinstalling by going to the the lunarvim install folder (not sure where it is on Mac, on linux it's usually ~/.local/share/lunarvim/lvim), then run `git switch release-1.2/neovim-0.8` then inside LunarVim sync plugins with `:PackerSync`
To go back to master `git switch master`

**Comment by opalmay** [comment]
> `black` works fine for me, you can set `lvim.lsp.null_ls.setup.debug = true`, restart lvim and format something then paste the log here (if it's long then as a file), the log is in `~/.cache/nvim/null-ls.log`, not `~/.cache/lvim` like `:NullLsLog` opens

Log showed
```
[TRACE Mon 16 Jan 2023 03:07:40 PM IST] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:204: output: nil
[TRACE Mon 16 Jan 2023 03:07:40 PM IST] ...ck/packer/start/null-ls.nvim/lua/null-ls/diagnostics.lua:181: received diagnostics from source 4
[TRACE Mon 16 Jan 2023 03:07:40 PM IST] ...ck/packer/start/null-ls.nvim/lua/null-ls/diagnostics.lua:182: {}
[TRACE Mon 16 Jan 2023 03:07:41 PM IST] .../site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:123: received LSP request for method textDocument/formatting
[TRACE Mon 16 Jan 2023 03:07:41 PM IST] ...ack/packer/start/null-ls.nvim/lua/null-ls/generators.lua:21: running generators for method NULL_LS_FORMATTING
[DEBUG Mon 16 Jan 2023 03:07:41 PM IST] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:319: spawning command "black" at /home/opal/Documents/projects/webAvailability_dev with args { "--stdin-filename", "/home/opal/Documents/projects/webAvailability_dev/lib/webMonitor.py", "--quiet", "--line-length", "200", "-" }
[TRACE Mon 16 Jan 2023 03:07:41 PM IST] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:203: error output: Traceback (most recent call last):
  File "/home/opal/.local/share/nvim/mason/bin/black", line 5, in <module>
    from black import patched_main
ModuleNotFoundError: No module named 'black'
```
Despite Mason showing them as installed. After uninstalling and reinstalling black through Mason issue was fixed.
Nullls info/checkhealth was showing fine because I had them in my path:
```
[opal@opal-nice ~]$ which black                                                                   
/usr/bin/black
[opal@opal-nice ~]$ which flake8
/usr/bin/flake8
```
However, it seems it didn't actually try using them

**Comment by opalmay** [comment]
> > File "/home/opal/.local/share/nvim/mason/bin/black", line 5, in 
> > from black import patched_main
> > ModuleNotFoundError: No module named 'black'
> 
> this looks like it was incorrectly installed
> 
> > However, it seems it didn't actually try using them
> 
> mason/bin is prepended to path so it the binaries there have higher priority
Mason was reporting them as installed. No idea what happened because I definitely did install and use them before through Mason.
Health checks probably reported ok because it was still in path, but actual formatting still tried to use Mason.

**Comment by imjamesdotme** [comment]
> 

Hey @opalmay 👋 Thanks for the information you've provided, much appreciated. I installed a fresh copy of `release-1.2/neovim-0.8` and added my config, everything worked as expected again.

For some reason `LvimUpdate` wasn't getting the latest changes so I ran the following `cd ~/.local/share/lunarvim/lvim && git checkout master` to get back to the latest changes.

I enabled debugging as @LostNeophyte suggested and tested out a file. Much to my suprise, everything was working still (`master-4b77c9c`).

There were no logs generated from this test but I could see one log from where I was running into issues this morning.

```
[WARN  Mon 16 Jan 09:43:50 2023] ...te/pack/lazy/opt/null-ls.nvim/lua/null-ls/generators.lua:94: failed to run generator: ...vim/site/pack/lazy/opt/null-ls.nvim/lua/null-ls/loop.lua:165: command eslint is not executable (make sure it's installed and on your $PATH)
```

It appears un-installing and re-installing has corrected whatever issue was occuring!
